packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -272,16 +347,33 @@ class UserSessionManager implements IUserSessionManager {
     value: Nullable<ApiObject> | Nullable<string> | undefined,
   ) {
     const entries = state.storage.entries.value;
-    const storage = entries[sessionKey]?.type as StorageType;
-    const key = entries[sessionKey]?.key as string;
-    if (isStorageTypeValidForStoringData(storage)) {
+    const storageType = entries[sessionKey]?.type as StorageType;
+    if (isStorageTypeValidForStoringData(storageType)) {
       const curStore = this.storeManager?.getStore(
-        storageClientDataStoreNameMap[storage] as string,
+        storageClientDataStoreNameMap[storageType] as string,
       );
-      if ((value && isString(value)) || isNonEmptyObject(value)) {
-        curStore?.set(key, value);
-      } else {
-        curStore?.remove(key);
+      if (curStore) {
+        const key = entries[sessionKey]?.key as string;
+        if (value && (isString(value) || isNonEmptyObject(value))) {
+          // if useServerSideCookies load option is set to true
+          // set the cookie from server side
+          if (
+            state.serverCookies.isEnabledServerSideCookies.value &&
+            storageType === COOKIE_STORAGE
+          ) {
+            this.setServerSideCookie([{ name: key, value }], curStore);
+          } else {
+            curStore?.set(key, value);
+          }
+        } else if (
+          state.serverCookies.isEnabledServerSideCookies.value &&
+          storageType === COOKIE_STORAGE
+        ) {
+          // remove cookie that is set from server side
+          this.setServerSideCookie([{ name: key, value: '' }], curStore, true);
```

Comment:
You don't need another flag `removeCookie` as you can use the empty value itself.

Reply:
Addressed